### PR TITLE
Openshift external secrets aws option

### PR DESCRIPTION
**Pull request by a1ex-var1amov**

### CODE REVIEW

File: roles/openshift_external_secrets/tasks/aws_backend.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+- name: Create AWS keys for user account which talks to AWS SM
+  k8s:
+    definition: "{{ lookup('template', 'aws-secret-manager-users-keys.yaml') }}"
+  when: external_secrets_oidc_enabled: false
```
Comment (noseka1) [inline roles/openshift_external_secrets/tasks/aws_backend.yaml]: Don't know if this syntax is correct. Should it be:
```
when: external_secrets_oidc_enabled
```
The change is required in multiple places.

File: roles/openshift_rhacm/tasks/delete_observability.yml [hunk 2]
```diff
@@ -18,9 +19,24 @@
       - pull-secret-secret.yaml
       - thanos-object-storage-secret.yaml
     kustomize_op: delete
+  when: rhacm_observability_use_existing_s3 == "false"
```
Comment (noseka1) [inline roles/openshift_rhacm/tasks/delete_observability.yml]: Comparing boolean to string. Should probably read:
```
when: not rhacm_observability_use_existing_s3
```

The change is required in multiple places.

**Review by noseka1 — CHANGES_REQUESTED** [review]
Thanks again for the pull request!

File: roles/openshift_external_secrets/tasks/aws_backend.yaml (hunk 1 above)
Comment (a1ex-var1amov) [reply roles/openshift_external_secrets/tasks/aws_backend.yaml]: Yup, it makes sense!

File: roles/openshift_external_secrets/tasks/deploy.yml [hunk 3]
```diff
@@ -1,7 +1,12 @@
-- import_tasks: deploy_operator.yml
+- include_tasks: deploy_operator.yml
 
-- import_tasks: deploy_instance.yml
+- include_tasks: deploy_instance.yml
 
-- import_tasks: vault_backend.yml
+- include_tasks: vault_backend.yml
+  when: external_secrets_secrets_manager_type == vault
```
Comment (noseka1) [inline roles/openshift_external_secrets/tasks/deploy.yml]: This should probably read:
```
when: external_secrets_secrets_manager_type == 'vault'
```
There are several more occurrences below.

**Review by noseka1 — CHANGES_REQUESTED** [review]
Thanks for the code updates! There is one more spot that looks suspicious.

**Comment by a1ex-var1amov** [comment]
Oh man. Ill double check

**Pull request merged**